lib/IRGen/GenDecl.cpp

```diff
@@ -3795,9 +3793,32 @@ void IRGenModule::appendLLVMUsedConditionalEntry(
                             llvm::ConstantAsMetadata::get(type),
                         }),
   };
+  UsedConditionals.push_back(llvm::MDNode::get(Module.getContext(), metadata));
+}
+
+void IRGenModule::emitUsedConditionals() {
+  if (UsedConditionals.empty())
+    return;
+
   auto *usedConditional =
       Module.getOrInsertNamedMetadata("llvm.used.conditional");
-  usedConditional->addOperand(llvm::MDNode::get(Module.getContext(), metadata));
+
+  for (auto M : UsedConditionals) {
```

Comment:
Is there a reason that  you didn't do `auto &M`?  This can potentially cause a copy .  Is this a pointer type?  In that case,  its easier to read as `auto *M`.

Reply:
Fixed.